src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php

```diff
@@ -129,16 +134,71 @@ private function handleConfig(ContainerBuilder $container, array $config, array
      *
      * @param ContainerBuilder $container
      * @param XmlFileLoader    $loader
+     * @param string[]         $bundles
      */
-    private function registerMetadataConfiguration(ContainerBuilder $container, XmlFileLoader $loader)
+    private function registerMetadataConfiguration(ContainerBuilder $container, XmlFileLoader $loader, array $bundles)
     {
-        $loader->load('metadata.xml');
+        $loader->load('metadata/metadata.xml');
+        $loader->load('metadata/xml.xml');
+
+        list($xmlResources, $yamlResources) = $this->getResourcesToWatch($container, $bundles);
+        $container->getDefinition('api_platform.metadata.extractor.xml')->addArgument($xmlResources);
+
+        if (class_exists(Annotation::class)) {
+            $loader->load('metadata/annotation.xml');
+        }
+
+        if (class_exists(Yaml::class)) {
+            $loader->load('metadata/yaml.xml');
+            $container->getDefinition('api_platform.metadata.extractor.yaml')->addArgument($yamlResources);
+        }
 
-        if (!interface_exists(DocBlockFactoryInterface::class)) {
-            $container->removeDefinition('api_platform.metadata.resource.metadata_factory.php_doc');
+        if (interface_exists(DocBlockFactoryInterface::class)) {
+            $loader->load('metadata/php_doc.xml');
```

Comment:
Why `php_doc` instead of `phpdoc`? For example the filename of the PSR draft is https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc.md